recipes/aws-libfabric/all/conanfile.py

```diff
@@ -0,0 +1,114 @@
+from conans import ConanFile, AutoToolsBuildEnvironment, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+required_conan_version = ">=1.35.0"
+
+class LibfabricConan(ConanFile):
+    name = "aws-libfabric"
+    description = "AWS Libfabric"
+    topics = ("fabric", "communication", "framework", "service")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/aws/libfabric"
+    license = "BSD-2-Clause", "GPL-2.0-or-later"
+    settings = "os", "arch", "compiler", "build_type"
+    _providers = ['gni', 'psm', 'psm2', 'sockets', 'rxm', 'tcp', 'udp', 'usnic', 'verbs', 'bgq', 'shm', 'efa', 'rxd', 'mrail', 'rstream', 'perf', 'hook_debug']
+    options = {
+        **{ p: [True, False, "shared"] for p in _providers },
+        **{
+            "shared": [True, False],
+            "fPIC": [True, False],
+            "with_libnl": [True, False],
+            "with_bgq_progress": ["auto", "manual"],
+            "with_bgq_mr": ["basic", "scalable"]
+        }
+    }
+    default_options = {
+        **{ p: False for p in _providers },
+        **{
+            "shared": False,
+            "fPIC": True,
+            "tcp": True,
+            "efa": tools.os_info.is_linux,
```

Comment:
Spot something here 🕵️ 

We are computing the packageIDs in a Linux node (we compute all the packageID before starting to build anything) and this `tools.os_info.is_linux` relies on the running machine. So here, it will compute a packageID that is different depending on the machine where you run `conan info` command.

The CI uses a profile for MacOS (running on a Linux node) and computes `package-id-A`, then it goes to the Macos node, it uses the same profile, but **the generated package-ID is different** (`efa` option changes default value). The command `conan upload` uses the full reference, with the package ID it is expecting:

```
conan upload .....:package-id-A
```

That concrete package-id is not available in that Macos node, because it uses a different `efa` option value and it has built a different package ID. Upload fails, package is not available in the node, it crashes.

`os_info` is really dangerous.

cc/ @danimtb @SSE4 @uilianries 

---

For the team: 

We need to check _again_ that the generated package-id is the one we are expecting to generate and raise a proper error _"package ID generated in build step is different from the one computed using 'conan info'"_, maybe add something more useful for the user: _"is there anything platform-dependent affecting the packageID (like `os_info`)?"_.

OTH, we might try to enforce some of this in `hooks`, but it would require monkey-patching of Conan client and it is probably too-much and undesired/risky.

---

For the PR (quick guess):

Probably this is the reason why some options list a valid `auto` value (additional to `True` and `False`), and it is the default one. That way the recipe can choose the value depending on the profile if it is not enforced. 

Something like:

```python
def _efa_enabled(self):
    if self.options.efa == "auto":
        return self.settings.os in ["Linux", ]. # <-- Depends on 'host' settings, not on the build machine.
    else:
        return self.options.efa
```

Reply:
I already removed the `auto` option because it hurts reproducibility. I've resolved to set the default in `configure()` instead.